### ISSUE TITLE
[10] BEIT: BERT Pre-Training of Image Transformers

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#37]
[paper](https://arxiv.org/pdf/2106.08254.pdf) [code](https://github.com/microsoft/unilm/tree/master/beit)

Image transformers 를 위한 pre-training 을 하자!
scratch train DeiT (81.8%) 보다 성능 좋음 (BEIT - 83.2%)
MIM (Masked Image Modeling) task 제안

## BEIT 
![image](https://user-images.githubusercontent.com/16400591/137618150-7b2d2d04-91af-4e03-90ae-345abaaa5011.png)

input image size (H, W)
patch size (P, P) 
일 때, 다음과 같은 형태로 일정하게 이미지를 나눔
![image](https://user-images.githubusercontent.com/16400591/137618393-ff75b5c3-6ad6-4f16-aafb-312382b072bc.png)

실험에서는 (224, 224) 를 16x16  patch 로 14x14 grid 를 만들었음
요 녀석들은 flatten 이후 encoder에 들어감

Image Tokenizer는 DALL-E의 dVAE(#38) 를 활용했음
tokenizer 는 image pixel을 discrete token z 로 mapping 시킴
deocder 는 z 에서 image pixel을 만들어줌
![image](https://user-images.githubusercontent.com/16400591/137618605-5d687ff7-3de1-4e4a-87ee-cc10300439eb.png)
![image](https://user-images.githubusercontent.com/16400591/137618611-d1e8c230-a2ef-4e95-8f1c-2c37e6233075.png)
visual vocab size |V| 는 8192 를 사용했고, 단순히 public 한 pretrained 모델을 사용했음

백본은 ViT 를 사용했음.
transformer input 들어갈 때, image patch들은 linear 하나 거침, 가장 앞에는 eos token [S] 를 붙여줌

## MIM (Masked Image Modeling)
Image patch 는 40% random mask
mask embedding e_[m] 은 learnable.

![image](https://user-images.githubusercontent.com/16400591/137619568-202c2e44-0f46-4487-89a2-fb1be0e8f67e.png)

## Details
12 layer transformer
random resized cropping, horizontal flipping, color jittering
ImageNet-1K (1.2M images)
500k steps (800 epoch)
2k batch-size
Adam (0.9, 0.999) , 1.5e-3 cosine lr, w-decay 0.05
warmup 10epoch
disable dropout
Stochastic Depth 사용 (0.1)
16 V100 5일
![image](https://user-images.githubusercontent.com/16400591/137620307-ed5b7da3-f649-48ea-8756-c498c1c48455.png)
![image](https://user-images.githubusercontent.com/16400591/137620313-425dcacb-9ec1-435f-bf0a-e6eba11617f4.png)
![image](https://user-images.githubusercontent.com/16400591/137620317-041740c7-affc-491a-a24b-8548b64db9de.png)

#### Image Classification
Image Classification 은 BEIT output 들에 GAP + softmax
![image](https://user-images.githubusercontent.com/16400591/137620088-4092108e-97be-4e1e-9d5f-70e2a39929db.png)
![image](https://user-images.githubusercontent.com/16400591/137620155-6328b397-4613-4a62-a362-922f209e8686.png)

당연하겠지만 수렴속도도 차이남
![image](https://user-images.githubusercontent.com/16400591/137620262-91737678-51d3-480a-b1f3-e0272eebc3c6.png)

#### Semantic Segmentation
Semantic Segmentation 은 SETR 사용 (BEIT encoder + deconv decoder)
ADE 20K에서 좋은 성능.
![image](https://user-images.githubusercontent.com/16400591/137620279-f67687a4-be5e-458d-b920-fec8203b31e1.png)

DINO 섞으면  ADE20K 에서 약간 더 좋아짐
![image](https://user-images.githubusercontent.com/16400591/137620341-d6934409-05b7-4ecb-a867-5e90997af539.png)

#### Ablation
![image](https://user-images.githubusercontent.com/16400591/137620449-9777edfe-7079-44e6-9153-439e9fe5170c.png)
